src/rpc/net.cpp

```diff
@@ -248,7 +248,16 @@ UniValue disconnectnode(const JSONRPCRequest& request)
     if(!g_connman)
         throw JSONRPCError(RPC_CLIENT_P2P_DISABLED, "Error: Peer-to-peer functionality missing or disabled");
 
-    bool ret = g_connman->DisconnectNode(request.params[0].get_str());
+    std::string node = (request.params[0].get_str());
+    bool ret;
+    if (std::all_of(node.begin(), node.end(), ::isdigit)) {
```

Comment:
> How about we add a new argument to this RPC called "nodeId"? The RPC can be called with strictly one of the arguments. There's no magic or ambiguity, and the node disconnection logic is contained within one RPC.

Sounds reasonable to me. To be clear: anything that doesn't encode the integer as a string is more reasonable to me than this. I don't have a strong opinion whether it should be one or multiple RPCs.